core/trino-main/src/main/java/io/trino/connector/system/SystemTablesProvider.java

```diff
@@ -30,4 +31,14 @@
      * can still be validly resolved.
      */
     Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName);
+
+    /**
+     * Resolves table name.
+     * In case that the system table supports versioning, a versioned representation of it
+     * will be returned.
+     */
+    default Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName, Optional<ConnectorTableVersion> startVersion, Optional<ConnectorTableVersion> endVersion)
+    {
+        return getSystemTable(session, tableName);
```

Comment:
Probably throw here. If a caller is expecting start and end version to be applied but the system table doesn't allow for it, the query should fail, not default to the non-versioned read.